File: Solutions/14581/java/program.java
// Copy from Quera

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);

        int n = scanner.nextInt();
        long sum = 0;
        int x = n / 2;
        sum = (long)(x - 1) * x / 2;
        sum *= 2;
        sum += x * (n % 2 + 1);
        double ans = (double)sum / (n + 1);
        System.out.printf("%.6f\n", ans);
    }
}

File: Solutions/14581/c++/program.cpp
// Copy from Quera

#include<iostream>
#include<iomanip>

using namespace std;

int main()
{
	int n;
	cin >> n;
	int k = n / 2;
	long long sum = 1ll * k * (k + 1) - (1 - n % 2) * k;
	cout << setprecision(6) << fixed << 1.0 * sum / (n + 1) << '\n';
	return 0;
}
